sigstore/_internal/oidc/__init__.py

```diff
@@ -69,3 +76,12 @@ def __init__(self, identity_token: str) -> None:
                 self.proof = str(identity_jwt["sub"])
             except KeyError:
                 raise IdentityError("Identity token missing `sub` claim")
+
+    def is_expired(self) -> bool:
+        """Verify if the identity token for this `Identity` is expired."""
+        if self.exp_timestamp:
+            now: float = datetime.now(timezone.utc).timestamp()
+            token_timestamp: float = self.exp_timestamp
+            return now > token_timestamp
+        else:
+            raise ValueError("Identity token does not have an expiration timestamp")
```

Comment:
Let's push this error state up to the constructor, and raise `IdentityError` (since it seems like a valid `Identity` _needs_ this claim.)